.travis.yml

```diff
@@ -7,32 +7,41 @@ matrix:
     - os: windows
       env: BUILD_CONFIGURATION=Release
     - os: linux
-      dist: trusty
-      group: travis_lts
-      sudo: required
-      env: BUILD_CONFIGURATION=Debug
+      compiler: gcc
+      addons: &gcc9
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['g++-9', 'gcc-9']
+      env: GCCVER='9' CXX_COMPILER=g++-${GCCVER} CXX=${CXX_COMPILER} C_COMPILER=gcc-${GCCVER} CC=${C_COMPILER} AR=gcc-ar-${GCCVER} RANLIB=gcc-ranlib-${GCCVER} NM=gcc-nm-${GCCVER} BUILD_CONFIGURATION=Release
     - os: linux
-      dist: trusty
```

Comment:
I think we should keep the `dist` name here. Of course Travis will pick some defaults for us but it would be good to explicitly specify which environment we use. Also, if Travis change the default bistro we will not get suddenly broken builds.

Reply:
my current best practice is to not specify the image for that reason, the default image is always a "stable" oldish one that is similar to what you can expect clients to have, but not *too* old. A breaking build on travis is not really an issue, it won't deploy, we will just know that we are using something that is now obsolete. We can set travis to build every month or so regardless of commits, so this kind of makes sure that we still run on something that is not obsolete. But the argument for "my travis didn't change, how did my build break" is also valid. 
So, as you prefer, we can put it back in, or even add a build without image specified (so that tests what happens when distros evolve), and one with "xenial" which is current default https://docs.travis-ci.com/user/reference/overview/
Anyway I'll follow whatever guideline you want, but personally I like the image-less version because it tests for obsolescence when travis update their setup (every year roughly). 
Like the current specified image was trusty, which is considered obsolete since april 2019

Reply:
I don't expect that something will be broken on the Travis side, but our build could fail because the environment has been changed silently. Also, it always gives a clear answer against which environment we validating.

>Like the current specified image was trusty, which is considered obsolete since April 2019

Thats ok and mostly indicate the lack of attention to the repo. Also, it explicitly guarantees that at least on trusty it works.

Reply:
To sum up: I suggest to set denial explicitly.

Reply:
ok